requirements.txt

```diff
@@ -1,5 +1,7 @@
 numpy
 scipy
+cvxpy
+cvxopt
```

Comment:
Aren't the `install_requirements` loaded from this file?

Reply:
No, these are specified independently in the `requirements.txt` file and the `setup.py`, as they serve two different purposes:

- The `requirements.txt` file is to allow developers to reproduce a single consistent environment when developing with the PennyLane source code. We actually are a bit lax here -- the `requirements.txt` file should have exact version pinning, however we avoid this at the moment for CI reasons.

- The `setup.py` file is the one read by `pip` before installation, so the opposite applies here; minimal pinning is preferred, to ensure that the package can be installed in a wide range of potential environments.